lxd/network_allocations.go

```diff
@@ -0,0 +1,224 @@
+//	          example: 200
+//	        metadata:
+//	          type: array
+//	          description: List of network allocations used by a consuming entity
+//	          items:
+//	            $ref: "#/definitions/NetworkAllocations"
+//	  "403":
+//	    $ref: "#/responses/Forbidden"
+//	  "500":
+//	    $ref: "#/responses/InternalServerError"
+func networkAllocationsGet(d *Daemon, r *http.Request) response.Response {
+	projectName, _, err := project.NetworkProject(d.State().DB.Cluster, projectParam(r))
+	if err != nil {
+		return response.SmartError(err)
+	}
+
+	allProjects := shared.IsTrue(queryParam(r, "all-projects"))
+
+	var projectNames []string
+	err = d.db.Cluster.Transaction(r.Context(), func(ctx context.Context, tx *db.ClusterTx) error {
+		// Figure out the projects to retrieve.
+		if !allProjects {
+			projectNames = []string{projectName}
+		} else {
+			// Get all project names if no specific project requested.
+			projectNames, err = dbCluster.GetProjectNames(ctx, tx.Tx())
+			if err != nil {
+				return fmt.Errorf("Failed loading projects: %w", err)
+			}
+		}
+
+		return nil
+	})
+	if err != nil {
+		return response.SmartError(err)
+	}
+
+	// Helper function to get the default CIDR address (/32 or /128 mask for ipv4 or ipv6 respectively)
+	getDefaultCIDRAddr := func(addr string) (string, error) {
+		ip := net.ParseIP(addr)
+		if ip.To4() != nil {
+			return fmt.Sprintf("%s/32", addr), nil
+		} else if ip.To16() != nil {
+			return fmt.Sprintf("%s/128", addr), nil
+		} else {
+			return "", fmt.Errorf("%s is not a valid IP address.", addr)
+		}
+	}
+
+	result := make([]api.NetworkAllocations, 0)
+
+	// Then, get all the networks, their network forwards and their network load balancers.
+	for _, projectName := range projectNames {
+		networkNames, err := d.db.Cluster.GetNetworks(projectName)
+		if err != nil {
+			return response.SmartError(err)
+		}
+
+		// Get all the networks, their attached instances, their network forwards and their network load balancers.
+		for _, networkName := range networkNames {
+			n, err := network.LoadByName(d.State(), projectName, networkName)
+			if err != nil {
+				return response.SmartError(err)
+			}
+
+			gwAddrs := make([]string, 0)
+			ipv4 := n.Config()["ipv4.address"]
+			ipv6 := n.Config()["ipv6.address"]
+			if ipv4 != "" {
+				gwAddrs = append(gwAddrs, ipv4)
+			}
+
+			if ipv6 != "" {
+				gwAddrs = append(gwAddrs, ipv6)
+			}
+
+			if len(gwAddrs) > 0 {
+				result = append(result, api.NetworkAllocations{
+					Addresses: gwAddrs,
+					UsedBy:    api.NewURL().Path(version.APIVersion, "networks", networkName).Project(projectName).String(),
+					Type:      "network",
+				})
+			}
+
+			leases, err := n.Leases(projectName, clusterRequest.ClientTypeNormal)
+			if err != nil {
+				return response.SmartError(err)
+			}
+
+			instanceAllocs := make(map[string]api.NetworkAllocations, 0)
+			for _, lease := range leases {
+				if lease.Type != "gateway" && lease.Type != "uplink" {
```

Comment:
Can we use this here?

```go
if shared.StringInSlice(lease.Type, []string{"static", "dynamic"}) {
```

To make it explicit which lease types we are including.